### PR TITLE
Update meta client interface and add skipConfig flag

### PR DESCRIPTION
**Pull request by dangleptr**
There are too many arguments for meta client now.  Some arguments are not required for each caller. 
Now we compact them into one structure "MetaClientOptions".  Meanwhile,  we add a new option to skip all config manager logic in it.  It is useful for many tools. 

By the way,  there are some problems for current heart beat logic after #1236  merged in. This pr has fixed it.

### CODE REVIEW

File: src/meta/client/MetaClient.cpp [hunk 1]
```diff
@@ -79,9 +73,11 @@ bool MetaClient::isMetadReady() {
 }
 
 bool MetaClient::waitForMetadReady(int count, int retryIntervalSecs) {
-    std::string gflagsJsonPath;
-    GflagsManager::getGflagsModule(gflagsModule_);
-    gflagsDeclared_ = GflagsManager::declareGflags(gflagsModule_);
+    if (!options_.skipConfig_) {
```
Comment (critical27) [inline src/meta/client/MetaClient.cpp]: 👍

File: src/storage/StorageServer.cpp [hunk 2]
```diff
@@ -89,11 +90,14 @@ bool StorageServer::start() {
     workers_->start();
 
     // Meta client
+    meta::MetaClientOptions options;
+    options.localHost_ = localHost_;
+    options.inStoraged_ = true;
+    options.serviceName_ = "";
```
Comment (darionyaphet) [inline src/storage/StorageServer.cpp]: why the service name is empty ?

Comment (dangleptr) [reply src/storage/StorageServer.cpp]: Ha,  it is compatible with current logic.  I did not handled it in this pr.

File: src/tools/storage-perf/StoragePerfTool.cpp [hunk 3]
```diff
@@ -74,7 +74,10 @@ class Perf {
             threads.emplace_back(std::move(t));
         }
         threadPool_ = std::make_shared<folly::IOThreadPoolExecutor>(FLAGS_io_threads);
-        mClient_ = std::make_unique<meta::MetaClient>(threadPool_, metaAddrsRet.value());
+
+        meta::MetaClientOptions options;
+        options.skipConfig_ = true;
+        mClient_ = std::make_unique<meta::MetaClient>(threadPool_, metaAddrsRet.value(), options);
```
Comment (CPWstatic) [inline src/tools/storage-perf/StoragePerfTool.cpp]: Would you please update the code in tools/db-dump/DbDumper.cpp ? It also initiates a meta client.

**Pull request merged**